# HG changeset patch
# User Christoph Reiter <reiter.christoph@gmail.com>
# Date 1410332863 -7200
# Node ID de134e036c45d99c08481c43a08955f31ea1dff7
# Parent  6342e2e6ed4bbb5bd1e2d6901bdc60ebb525399f
Fix normalize_path() on darwin. Closes issue 1465.

diff --git quodlibet/util/path.py.orig quodlibet/util/path.py
--- quodlibet/util/path.py.orig
+++ quodlibet/util/path.py
@@ -343,7 +343,8 @@
     return drive + tail
 
 
-def _normalize_darwin_path(filename, strict=False, _cache={}, _statcache={}):
+def _normalize_darwin_path(filename, canonicalise=False, strict=False,
+                           _cache={}, _statcache={}):
     """Get a normalized version of the path by calling listdir
     and comparing the inodes with our file.
 
@@ -352,6 +353,9 @@
     - Supports relative and absolute paths (and returns the same).
     """
 
+    if canonicalise:
+        filename = os.path.realpath(filename)
+
     if filename in (".", "..", "/", ""):
         return filename
 
diff --git tests/test_util.py.orig tests/test_util.py
--- tests/test_util.py
+++ tests/test_util.py
@@ -695,8 +695,11 @@
             os.rmdir(name)
 
     def test_canonicalise(self):
-        from quodlibet.util.path import _normalize_path as norm
+        from quodlibet.util.path import _normalize_path, _normalize_darwin_path
+        self._test_canonicalise(_normalize_path)
+        self._test_canonicalise(_normalize_darwin_path)
 
+    def _test_canonicalise(self, norm):
         f, path = tempfile.mkstemp()
         os.close(f)
         path = norm(path)
